### PR TITLE
Test for bundlepaths before deprecate

### PR DESCRIPTION
**Pull request by ankitathomas**
Allow deprecatetruncate to handle multiple bundles on the same upgrade chain. Only the most latest bundle in the chain will be marked as deprecated, the remaining will be truncated with the rest of the upgrade chain older than the deprecated bundle.

### CODE REVIEW

File: pkg/sqlite/deprecate_test.go [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+			querier := NewSQLLiteQuerierFromDb(db)
+
+			bundleLoader := registry.BundleGraphLoader{}
+			for _, b := range tt.initial {
+				graph, err := graphLoader.Generate(b.pkg)
+				if err != nil {
+					require.EqualError(t, err, registry.ErrPackageNotInDatabase.Error(), "Expected empty error or package not in database")
+				}
+				bndl := newRegistryBundle(t, b)
+				graph, err = bundleLoader.AddBundleToGraph(bndl, graph, &registry.AnnotationsFile{Annotations: *bndl.Annotations}, false)
```
Comment (hasbro17) [inline pkg/sqlite/deprecate_test.go:317]: Missing error check?
```suggestion
				graph, err = bundleLoader.AddBundleToGraph(bndl, graph, &registry.AnnotationsFile{Annotations: *bndl.Annotations}, false)
				require.NoError(t, err)
```

**Comment by hasbro17** [comment]
SGTM overall.

File: pkg/sqlite/deprecate_test.go [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/blang/semver/v4"
+	"github.com/operator-framework/api/pkg/lib/version"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/operator-registry/pkg/api"
+	image2 "github.com/operator-framework/operator-registry/pkg/image"
```
Comment (timflannagan) [inline pkg/sqlite/deprecate_test.go:17]: Minor nit, but any reason why we need to rename this import declaration and can't just use the regular path?

File: pkg/sqlite/deprecate_test.go [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+				graph, err = bundleLoader.AddBundleToGraph(bndl, graph, &registry.AnnotationsFile{Annotations: *bndl.Annotations}, false)
+				require.NoError(t, store.AddBundleSemver(graph, bndl))
+			}
+			err = NewSQLDeprecatorForBundles(store, querier, tt.deprecate).Deprecate()
+			if tt.expectedError != nil {
+				require.EqualError(t, err, tt.expectedError.Error())
+				return
+			}
+			require.NoError(t, err)
+			result, err := querier.ListBundles(context.TODO())
```
Comment (timflannagan) [inline pkg/sqlite/deprecate_test.go:326]: Do we need to also check that this error is not nil?

File: pkg/sqlite/deprecate_test.go (hunk 1 above)
Comment (timflannagan) [reply pkg/sqlite/deprecate_test.go:317]: Yeah, agreed we should catch this error as well as the store.AddBundleSemver.

File: pkg/sqlite/deprecate_test.go (hunk 3 above)
Comment (ankitathomas) [reply pkg/sqlite/deprecate_test.go:317]: That's fair, I can take out that check

**Review by hasbro17 — COMMENTED** [review]
/lgtm

**Pull request merged**